### PR TITLE
Fix Golint errors in pkg/controller/podautoscaler

### PR DESCRIPTION
**Pull request by alexzimmer96**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR fixes golint errors inside pkg/controller/podautoscaler.

**Which issue(s) this PR fixes**:
Part of #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
/approve
for hack/

File: pkg/controller/podautoscaler/rate_limiters.go [hunk 1]
```diff
@@ -29,20 +29,25 @@ type FixedItemIntervalRateLimiter struct {
 
 var _ workqueue.RateLimiter = &FixedItemIntervalRateLimiter{}
 
+// NewFixedItemIntervalRateLimiter creates a new instance of an RateLimiter using a fixed interval
```
Comment (neolit123) [inline pkg/controller/podautoscaler/rate_limiters.go]: `of a RateLimiter...`

File: pkg/controller/podautoscaler/rate_limiters.go [hunk 2]
```diff
@@ -29,20 +29,25 @@ type FixedItemIntervalRateLimiter struct {
...
 func NewFixedItemIntervalRateLimiter(interval time.Duration) workqueue.RateLimiter {
 	return &FixedItemIntervalRateLimiter{
 		interval: interval,
 	}
 }
 
+// When gets an item and gets to decide how long that item should wait
```
Comment (neolit123) [inline pkg/controller/podautoscaler/rate_limiters.go]: > When returns the interval of the rate limiter

File: pkg/controller/podautoscaler/rate_limiters.go [hunk 3]
```diff
@@ -29,20 +29,25 @@ type FixedItemIntervalRateLimiter struct {
...
 	return r.interval
 }
 
+// NumRequeues returns back how many failures the item has had
 func (r *FixedItemIntervalRateLimiter) NumRequeues(item interface{}) int {
 	return 1
 }
 
+// Forget indicates that an item is finished being retried.  Doesn't matter whether its for perm failing
+// or for success, we'll stop tracking it
```
Comment (neolit123) [inline pkg/controller/podautoscaler/rate_limiters.go]: possibly omit the second sentence.

**Review by neolit123 — COMMENTED** [review]
thanks @alexzimmer96 ,
added some minor nits around function comments.

**Comment by alexzimmer96** [comment]
hi @neolit123, i've added your suggestions.
Thanks for reviewing

**Review by neolit123 — COMMENTED** [review]
/lgtm
thanks for the updates.

**Review by SataQiu — APPROVED** [review]
/LGTM

**Comment by alexzimmer96** [comment]
What is this PR hanging on ATM? Can i do anything to shorten the process?

**Comment by BenTheElder** [comment]
> What is this PR hanging on? Can i do anything to shorten the process?

see at the bottom:
> tide Pending — Not mergeable. Needs approved label.

we're in code freeze. a notice was sent to the kubernetes-dev mailinglist.
while I don't like this process much myself, only urgent fixes labeled for the current release milestone are merged during code freeze. it will merge after code freeze.

schedule is here https://github.com/kubernetes/sig-release/tree/master/releases/release-1.18

**Comment by BenTheElder** [comment]
oh and someone from https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/OWNERS still needs to approve it looks like (which is why there's no "approved" label)

@MaciekPytel may not be active on on this (@kubernetes/sig-contributor-experience is looking to clean these entries up...)
/assign piosz

**Pull request merged**